dataplane/internal/kernel/kerneltest/kernel.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// package kerneltest contains fake implemetation of structs in kernel package
+package kerneltest
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/vishvananda/netlink"
+)
+
+// Iface is a basic representation of a network interface.
+type Iface struct {
+	HWAddr net.HardwareAddr
+	Idx    int
+	IPs    map[string]struct{}
+	up     bool
+}
+
+// New returns a new FakeInterfaces.
+func New() *FakeInterfaces {
+	return &FakeInterfaces{
+		Links: map[string]*Iface{},
+	}
+}
+
+// FakeInterfaces contains a fake implementation methods for modifying networking interfaces.
+type FakeInterfaces struct {
```

Comment:
This file looks to me more like an interface manager API instead of true fake interfaces since you're calling netlink and plugging in channel objects that interface with the netlink package. Would it be more appropriate to just call it InterfaceManager? I know it's the same name as the interface type but I think that shouldn't influence the struct's name.

Reply:
no netlink calls are made, they are all faked out. just reusing the types. it is a fake

Reply:
I see, I thought that was a function call, thakns for changing it